src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -1021,14 +1035,26 @@ case class MatrixMapRows(child: MatrixIR, newRow: IR, newKey: Option[(IndexedSeq
 
     val colValuesType = TArray(prev.typ.colType)
     val vaType = prev.typ.rvRowType
-    val (rvAggs, initOps, seqOps, aggResultType, f, rTyp) = ir.CompileWithAggregators[Long, Long, Long, Long, Long, Long](
+
+    var scanInitNeedsGlobals = false
```

Comment:
yeah. Some of the other nodes aren't copying globals/col values where they aren't needed, so I put it in here as well. I figure since there are so many stages where the globals have to be re-copied, it would probably be worth it, but if you think it adds more complexity than anything else I'm happy to take it out.